### ISSUE TITLE
The usage of BEiT_win

### ISSUE THREAD

**Issue by ydhongHIT** [baaivision/EVA#32]
Hi, it is very kind to provide the BEiT with window attention. Did you try this model in seg for more efficient memory usage and how about the accuracy? In addition, you seem to drop the cls embedding and change the number of relative position bias from (2*Wh-1 * 2*Ww-1 + 3) to (2*Wh-1 * 2*Ww-1). So the relation position bias will be initialized randomly rather than loaded from pre-trained weights?

**Comment by Yuxin-CV** [comment]
In our study, we find win_attn is more beneficial to object detection tasks, we also observe slightly degenerated performance using win_attn in sem seg. Actually, different from object detection, sem seg can hardly benefit from very large resolution inputs. 

During MIM pt, we don't use rpe. So rpe will always be randomly initialized in ft.

**Comment by ydhongHIT** [comment]
Thanks for your quick reply.

**Comment by ydhongHIT** [comment]
> In our study, we find win_attn is more beneficial to object detection tasks, we also observe slightly degenerated performance using win_attn in sem seg. Actually, different from object detection, sem seg can hardly benefit from very large resolution inputs.
> 
> During MIM pt, we don't use rpe. So rpe will always be randomly initialized in ft.

Hi, could you provide the detailed results for "we also observe slightly degenerated performance using win_attn in sem seg"? In my experiments, the performance degradation is about 2-3 points.

**Comment by Yuxin-CV** [comment]
> > In our study, we find win_attn is more beneficial to object detection tasks, we also observe slightly degenerated performance using win_attn in sem seg. Actually, different from object detection, sem seg can hardly benefit from very large resolution inputs.
> > During MIM pt, we don't use rpe. So rpe will always be randomly initialized in ft.
> 
> Hi, could you provide the detailed results for "we also observe slightly degenerated performance using win_attn in sem seg"? In my experiments, the performance degradation is about 2-3 points.

Our observations are the same with you: we also suffer from around 2 to 3 points performance degeneration.

**Comment by ydhongHIT** [comment]
> > > In our study, we find win_attn is more beneficial to object detection tasks, we also observe slightly degenerated performance using win_attn in sem seg. Actually, different from object detection, sem seg can hardly benefit from very large resolution inputs.
> > > During MIM pt, we don't use rpe. So rpe will always be randomly initialized in ft.
> > 
> > 
> > Hi, could you provide the detailed results for "we also observe slightly degenerated performance using win_attn in sem seg"? In my experiments, the performance degradation is about 2-3 points.
> 
> Our observations are the same with you: we also suffer from around 2 to 3 points performance degeneration.

Thanks. In fact, I don't understand why the performance degeneration is so obvious. Since the window size is equal to that used in the pre-training stage (e.g. 14x14), it should generalize to downstream tasks.

**Comment by Yuxin-CV** [comment]
Initially, I was also perplexed by this. However, I now believe this relates to the task: semantic segmentation does not require exceptionally high resolution inputs to achieve state-of-the-art performance. For example, in EVA-02, we found 512px and 640px inputs to be adequate, while 896px did not improve performance. Under these circumstances, I do not believe using windowed attention would yield further enhancements. 

I also comprehend your concern that during fine-tuning, if the window size equals that used in pre-training, pre-training and fine-tuning would be better aligned. However, we found using fully global attention or more global attention achieved comparable performance to the default settings for ViTDet.

**Comment by ydhongHIT** [comment]
> 

I get it. I agree with you that semantic segmentation may not require exceptionally high resolution inputs and using fully global attention is usually better. But I think using local window attention is still better sometimes. On the one hand, the input image is inevitably large. For example, Cityscapes consists of images with large resolution. On the other hand, using local window attention can accelerate the train and inference under 640x640 resolution. I think adapting global attention to local directly is convenient and elegant. However, it may not be suitable for semantic segmentation task at the moment. Anyway, thanks again for your patient response.

**Comment by ydhongHIT** [comment]
> Have you tried sem seg on Cityscapes with win-attn? I think it will work.
> 
> For the speed and memory concerns with global-attn, I think xFormer / Flash attn / SDPA in pytorch 2.0+ will help.

I do not try it on Cityscapes. Do you mean that win-attn will bring litte performance degeneration on this occasion? Maybe less than one point?

**Comment by Yuxin-CV** [comment]
> > Have you tried sem seg on Cityscapes with win-attn? I think it will work.
> > For the speed and memory concerns with global-attn, I think xFormer / Flash attn / SDPA in pytorch 2.0+ will help.
> 
> I do not try it on Cityscapes. Do you mean that win-attn will bring litte performance degeneration on this occasion? Maybe less than one point?


I concur. However, to my knowledge, the results of semantic segmentation on Cityscapes frequently fluctuate substantially; this is tedious to study. :disappointed:

**Comment by ydhongHIT** [comment]
>
